src/test/groovy/com/streamr/client/StreamrClientSpec.groovy

```diff
@@ -0,0 +1,135 @@
+package com.streamr.client
+
+import com.streamr.client.options.ResendLastOption
+import com.streamr.client.options.SigningOptions
+import com.streamr.client.options.StreamrClientOptions
+import com.streamr.client.protocol.control_layer.BroadcastMessage
+import com.streamr.client.protocol.control_layer.ResendLastRequest
+import com.streamr.client.protocol.control_layer.ResendRangeRequest
+import com.streamr.client.protocol.control_layer.ResendResponseResent
+import com.streamr.client.protocol.control_layer.SubscribeRequest
+import com.streamr.client.protocol.control_layer.SubscribeResponse
+import com.streamr.client.protocol.control_layer.UnicastMessage
+import com.streamr.client.protocol.message_layer.MessageID
+import com.streamr.client.protocol.message_layer.MessageRef
+import com.streamr.client.protocol.message_layer.StreamMessage
+import com.streamr.client.protocol.message_layer.StreamMessageV30
+import com.streamr.client.rest.Stream
+import spock.lang.Specification
+
+class StreamrClientSpec extends Specification {
+
+    private static TestWebSocketServer server = new TestWebSocketServer("localhost", 6000)
+
+    TestingStreamrClient client
```

Comment:
Hmm.. I wonder would it be better to use `StreamrClient` directly instead of using a derived class. I believe you could invoke `StreamrClient#handleMessage` and access `StreamrClient.subs` from this test directly since they are protected and this test and StreamrClient live in the same package.

Reply:
This class was created to be able to access the `handleMessage` and `subs` without making them `public` in `StreamrClient` since this is only for testing and shouldn't affect encapsulation too much (still had to make them `protected` instead of `private`).

Reply:
As long as the subclass isn't stubbing out behavior I think it is fine. As is the case here.